openapi_configuration.yml

```diff
@@ -540,41 +540,199 @@ paths:
+      summary: Extract from a reference document using the specified published configuration. 
+      description: Extract from a reference document using the specified published configuration
+      tags:
+      - Reference document exraction
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PostGoldenExtraction'
+      responses:
+          '200':
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/ResponseGoldenExtraction'
+            description: The structured data extracted from the reference document
+
+          400: # todo I didn't verify 40x responses for any extract_from_goldens, just assumed
+            $ref: '#/components/responses/400'
+          401:
+            $ref: '#/components/responses/401'
+          404:
+            $ref: '#/components/responses/404'
+          415:
+            $ref: '#/components/responses/415'
+          500:
+            $ref: '#/components/responses/500' 
+
+  /extract_from_golden/{type}/{name}/{version}/{document-name}: # question this one is pretty confusing to me. why is the golden in the path instead of body like the others? why is it a get not post? I didn't dig deep
+    parameters:
+      - $ref: '#/components/parameters/documentTypeName'
+      - $ref: '#/components/parameters/configurationName'
+      - $ref: '#/components/parameters/versionIdentifier'
+      - $ref: '#/components/parameters/documentName'
+      - $ref: '#/components/parameters/environment' 
+    get: 
+      operationId: extract-from-reference-document-and-config-version
+      summary: Extract from a reference document using the specified version of a configuration
+      description: tbd
+      tags:
+      - Reference document exraction
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ResponseGoldenExtraction' # todo it's a get, figure out the request and response which I think is different from other responses..?...
+      responses:
+          '200':
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/ResponseGoldenExtraction'
+            description: The structured data extracted from the reference document 
+          400: # todo I didn't verify 40x responses for any extract_from_goldens, just assumed
+            $ref: '#/components/responses/400'
+          401:
+            $ref: '#/components/responses/401'
+          404:
+            $ref: '#/components/responses/404'
+          415:
+            $ref: '#/components/responses/415'
+          500:
+            $ref: '#/components/responses/500'
+
+  /extract_text_from_golden/{type}: # question: configuration: configurationSchema doesn't make sense in the body request here but it looks like it's allowed?
```

Comment:
if a configuration is supplied, the text extraction uses the defined preprocessors.

i wasn't part of the discussion around this but my guess is that we accept a full configuration for convenience, users can just pass in the configuration as a whole if they wish to make use of this feature